pandas/io/formats/style_render.py

```diff
@@ -847,23 +858,25 @@ def _translate_latex(self, d: dict, clines: str | None) -> None:
             for r, row in enumerate(d["head"])
         ]
 
-        def concatenated_visible_rows(obj, n, row_indices):
+        def _concatenated_visible_rows(obj, n, row_indices):
             """
             Extract all visible row indices recursively from concatenated stylers.
             """
             row_indices.extend(
                 [r + n for r in range(len(obj.index)) if r not in obj.hidden_rows]
             )
-            return (
-                row_indices
-                if obj.concatenated is None
-                else concatenated_visible_rows(
-                    obj.concatenated, n + len(obj.index), row_indices
-                )
-            )
+            n += len(obj.index)
+            for concatenated in obj.concatenated:
+                n = _concatenated_visible_rows(concatenated, n, row_indices)
+            return n
+
+        def concatenated_visible_rows(obj):
+            row_indices: list[int] = []
+            _concatenated_visible_rows(obj, 0, row_indices)
```

Comment:
Can this function (and the one above) be refactored to just recursively yield each row, since it's just needed in the `zip` below? From first glance, it seems odd that this is modifying `row_indices` inplace and the result `n` isn't used

Reply:
I agree this could be considered for refactor to be made simpler. Does it have to be a blocker for the PR - it is otherwise a very nice addition that addresses a potentially common use case?

Reply:
Sure can be a follow up but it would be good to add a TODO comment here with the note to simplify